ctera/src/main/java/ch/cyberduck/core/ctera/CteraMoveFeature.java

```diff
@@ -20,19 +20,45 @@
 import ch.cyberduck.core.dav.DAVMoveFeature;
 import ch.cyberduck.core.exception.BackgroundException;
 import ch.cyberduck.core.exception.InvalidFilenameException;
+import ch.cyberduck.core.exception.NotfoundException;
 
 import java.text.MessageFormat;
 
+import static ch.cyberduck.core.ctera.CteraAttributesFinderFeature.*;
+
 public class CteraMoveFeature extends DAVMoveFeature {
 
+    private final CteraAttributesFinderFeature attributes;
+
     public CteraMoveFeature(final CteraSession session) {
         super(session);
+        this.attributes = new CteraAttributesFinderFeature(session);
+    }
+
+    public CteraMoveFeature(final CteraSession session, final CteraAttributesFinderFeature attributes) {
+        super(session);
+        this.attributes = attributes;
     }
 
     @Override
     public void preflight(final Path source, final Path target) throws BackgroundException {
         if(!CteraTouchFeature.validate(target.getName())) {
             throw new InvalidFilenameException(MessageFormat.format(LocaleFactory.localizedString("Cannot rename {0}", "Error"), source.getName())).withFile(source);
         }
+        assumeRole(source, DELETEPERMISSION);
+        boolean targetExists = true;
+        try {
+            attributes.find(target);
```

Comment:
Long running operations are not allowed in preflight checks.

Reply:
03b8b3d2fec00a5d57763c3470018a0035d5e2f0